security/entity_provider.rst

```diff
@@ -108,13 +110,13 @@ with the following fields: ``id``, ``username``, ``password``,
         public function eraseCredentials()
         {
         }
-
+        
         /** @see \Serializable::serialize() */
         public function serialize()
         {
             return serialize(array(
                 $this->id,
-                $this->username,
+                $this->username, // you should use $this->email if you don't use username but email to log user
```

Comment:
IMO we should remove this comment. Similar things apply to all other attributes as well if you change their names.